base64ct/src/lib.rs

```diff
@@ -287,28 +277,25 @@ pub const fn decoded_len(input: &str, padded: bool) -> usize {
             i -= 1;
         }
 
+        // TODO(tarcieri): fix potential overflow
         ((bytes.len() - pad_count) * 3) / 4
     } else {
-        decoded_len_inner_unpadded(input.len())
+        // branchless, overflow-proof computation of `(3*n)/4`
+        let n = input.as_ref().len();
+        let k = n / 4;
+        let l = n - 4 * k;
+        3 * k + (3 * l) / 4
     }
 }
 
-#[inline(always)]
-const fn decoded_len_inner_unpadded(n: usize) -> usize {
-    // branchless, overflow-proof computation of `(3*n)/4`
-    let k = n / 4;
-    let l = n - 4 * k;
-    3 * k + (3 * l) / 4
-}
```

Comment:
We could potentially keep this extracted as a separate function, but as of now the corresponding padded implementation isn't `const`-friendly.

It would be possible to support a fully const-friendly implementation, but the padded implementation needs to be reworked.